src/Logic.hs

```diff
@@ -428,18 +445,25 @@ handleCommentAdded triggerConfig projectConfig mergeWindowExemption prId author
             let (Config.MergeWindowExemptionConfiguration users) = mergeWindowExemption
             in elem user users
       isApproved <-
-        if isJust approvalType
+        if isSuccess approvalType
           then isReviewer author
           else pure False
       -- For now Friday at UTC+0 is good enough.
       -- See https://github.com/channable/hoff/pull/95 for caveats and improvement ideas.
       day <- dayOfWeek . utctDay <$> getDateTime
-      if isApproved
-        then -- The PR has now been approved by the author of the comment.
-         case fromJust approvalType of
-           -- To guard against accidental merges we make use of a merge window.
-           -- Merging inside this window is discouraged but can be overruled with a special command or by adding the
-           -- user to the merge window exemption list.
+      case approvalType of
+        -- The bot was not mentioned in the comment, ignore
+        Ignored -> pure state
+        -- The bot was mentioned but encountered an invalid command, report error and
+        -- take not further action
+        Unknown command -> do
+          () <- leaveComment prId ("`" <> command <> "` was not recognized as a valid command.")
+          pure state
+        -- Cases where the parse was successful AND the author is a reviewer
+        Success command | isApproved -> case command of
```

Comment:
Personally, I find this guard a bit subtle for the security check. Could you make it an explicit `if`, that way changing the order of the guards doesn't become a security concern.

Reply:
Alternatively, you could attach both guards to the same branch, i.e.

```haskell
case .. of
  ...
  Success command
    | isApproved -> ...
    | otherwise -> ...
```

which IMO is the nicer style even in the absence of security concerns, as it makes it very clear which sub-cases there are for the `Success` case.

---

Also, `isApproved` seems to be the conjunction of the success case and the author-is-reviewer case. But you're already checking the success case by explicit pattern matching now. So just checking for `isReviewer` in the guard would be even cleaner.

Reply:
Thanks, I think I'll adopt the style mentioned by @fatho :). 

Regarding `isReviewer`, putting it in the guard makes sense, but the function returns `Action`, so it can't be done within the guard itself (as far as I know), so doing `isSuccess` and also pattern matching on the `Success` case is a bit of a compromise to keep the bot from performing an `Action` every time it reads a comment regardless of whether it parsed a command. :/